server/filestore.go

```diff
@@ -5941,6 +6107,7 @@ func (mb *msgBlock) indexCacheBuf(buf []byte) error {
 		seq = le.Uint64(hdr[4:])
 
 		// Clear any headers bit that could be set.
+		hasHeaders := rl&hbit != 0
```

Comment:
This does not mark headers, it marks if we have checked the hash checksum or not.

Reply:
I looked into this after we spoke yesterday and found it actually has two purposes. When applied to the `mb.cache.idx` it's for the hash checksum, when applied to the `rl` it signifies the presence of a header. 

The `hbit` definition also says about this, although it's a bit vague about the difference:
```
	// Used for marking messages that have had their checksums checked.
	// Used to signal a message record with headers.
	hbit = 1 << 31
```

Reply:
My apologies. I wonder if we should have a new constant (same value) that allows us to distinguish?